src/ripple/app/tests/MultiSign.test.cpp

```diff
@@ -394,6 +412,209 @@ class MultiSign_test : public beast::unit_test::suite
+            Json::Value jv_one = setup_tx();
+            jv_one[jss::tx_json][jss::Fee] = -1;
+            cheri_sign(jv_one);
+            auto jrr = env.rpc("json", "sign_for", jv_one.toStyledString())[jss::result];
+            BEAST_EXPECT(jrr[jss::status] == "success");
+
+            // for the second sign_for, use the returned tx_json with
+            // first signer info
+            Json::Value jv_two;
+            jv_two[jss::tx_json] = jrr[jss::tx_json];
+            becky_sign(jv_two);
+            jrr = env.rpc("json", "sign_for", jv_two.toStyledString())[jss::result];
+            BEAST_EXPECT(jrr[jss::status] == "success");
+
+            Json::Value jv_submit;
+            jv_submit[jss::tx_json] = jrr[jss::tx_json];
+            jrr = env.rpc("json", "submit_multisigned", jv_submit.toStyledString())[jss::result];
+            BEAST_EXPECT(jrr[jss::status] == "error");
+            BEAST_EXPECT(jrr[jss::error]  == "invalidParams");
+            BEAST_EXPECT(jrr[jss::error_message]  == "Invalid Fee field.  Fees must be greater than zero.");
+        }
+
+        {
+            //failure case - bad fee v2
+            aliceSeq = env.seq (alice);
+            Json::Value jv_one = setup_tx();
+            jv_one[jss::tx_json][jss::Fee]  = alice["USD"](10).value().getFullText();
+            cheri_sign(jv_one);
+            auto jrr = env.rpc("json", "sign_for", jv_one.toStyledString())[jss::result];
+            BEAST_EXPECT(jrr[jss::status] == "success");
+
+            // for the second sign_for, use the returned tx_json with
+            // first signer info
+            Json::Value jv_two;
+            jv_two[jss::tx_json] = jrr[jss::tx_json];
+            becky_sign(jv_two);
+            jrr = env.rpc("json", "sign_for", jv_two.toStyledString())[jss::result];
+            BEAST_EXPECT(jrr[jss::status] == "success");
+
+            Json::Value jv_submit;
+            jv_submit[jss::tx_json] = jrr[jss::tx_json];
+            jrr = env.rpc("json", "submit_multisigned", jv_submit.toStyledString())[jss::result];
+            BEAST_EXPECT(jrr[jss::status] == "error");
+            BEAST_EXPECT(jrr[jss::error]  == "internal");
+            BEAST_EXPECT(jrr[jss::error_message]  == "Internal error.");
+        }
+
+        {
+            // cheri should not be able to multisign using her master key.
+            aliceSeq = env.seq (alice);
+            Json::Value jv = setup_tx();
+            jv[jss::account]                       = cheri.human();
+            jv[jss::secret]                        = cheri.name();
+            auto jrr = env.rpc("json", "sign_for", jv.toStyledString())[jss::result];
+            BEAST_EXPECT(jrr[jss::status] == "error");
+            BEAST_EXPECT(jrr[jss::error]  == "masterDisabled");
+            env.close();
+            BEAST_EXPECT(env.seq(alice) == aliceSeq);
+        }
+
+        {
+            // becky should be also be able to sign using either her master key
```

Comment:
Text in comment?  Perhaps: "Unlike cheri, becky should be able to sign using her master key"?